netaddr.go

```diff
@@ -1245,6 +1248,68 @@ func (r IPRange) contains(addr IP) bool {
 	return r.From.Compare(addr) <= 0 && r.To.Compare(addr) >= 0
 }
 
+func (r IPRange) less(other IPRange) bool {
+	if cmp := r.From.Compare(other.From); cmp != 0 {
+		return cmp < 0
+	}
+	return r.To.Less(other.To)
+}
+
+// mergeIPRanges returns the minimum and sorted set of IP ranges that
+// cover r.
+func mergeIPRanges(r []IPRange) (out []IPRange, valid bool) {
```

Comment:
for a slice I'd use `rr`, `rs`, or `rv` , then you can use `r` as your iteration variable on line 1273